src/vs/workbench/contrib/chat/common/chatWordCounter.ts

```diff
@@ -10,13 +10,43 @@ export interface IWordCountResult {
 	isFullString: boolean;
 }
 
+const r = String.raw;
+
+/**
+ * Matches `[text](link title?)` or `[text](<link> title?)`
+ *
+ * Taken from vscode-markdown-languageservice
+ */
+const linkPattern =
```

Comment:
I feel like ideally we'd use a proper markdown parser here but this works fairly well